File: Raven.Client.Lightweight/Listeners/IDocumentDeleteListener.cs
//-----------------------------------------------------------------------
// <copyright file="IDocumentDeleteListener.cs" company="Hibernating Rhinos LTD">
//     Copyright (c) Hibernating Rhinos LTD. All rights reserved.
// </copyright>
//-----------------------------------------------------------------------
using Raven.Json.Linq;

namespace Raven.Client.Listeners
{
	/// <summary>
	/// Hook for users to provide additional logic on delete operations
	/// </summary>
	public interface IDocumentDeleteListener
	{
		/// <summary>
		/// Invoked before the delete request is sent to the server.
		/// </summary>
		/// <param name="key">The key.</param>
		/// <param name="entityInstance">The entity instance.</param>
		/// <param name="metadata">The metadata.</param>
		void BeforeDelete(string key, object entityInstance, RavenJObject metadata);
	}
}

File: Raven.Client.Java/src/main/java/net/ravendb/client/listeners/IDocumentDeleteListener.java
package net.ravendb.client.listeners;

import net.ravendb.abstractions.json.linq.RavenJObject;

/**
 * Hook for users to provide additional logic on delete operations
 */
public interface IDocumentDeleteListener {

  /**
   * Invoked before the delete request is sent to the server.
   * @param key The key.
   * @param entityInstance The entity instance.
   * @param metadata The metadata.
   */
  public void beforeDelete(String key, Object entityInstance, RavenJObject metadata);
}
